### PR TITLE
chore: update to @patternfly/pfe-core version 3.0

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1. Updated and moved `@patternfly/elements` to devDependencies at version `^3.0.0` 
2. Added `@patternfly/pfe-core` at `^3.0.0` to dependencies
3. Updated `lit` to `^3.1.2`
4. Updated `tslib` to `^2.6.2`
5. Updated `typescript` to  `^5.4.3` 
6. Switch `ttypescript` to `ts-patch`, replacing `ttsc` with `tspc`
7. Updated typescript config `tsconfig` to match expected output.
8. Update api's for InternalController and RTI where necessary.

## Testing Instructions

1.

### CODE REVIEW

File: package.json
Comment (bennypowers) [inline package.json:222]: I want this to be a dev dependency, or to be removed entirely

We should only need core and tools

**Review by nikkimk — APPROVED** [review]
Littered gymnasium: total mayhem

**Review by nikkimk — APPROVED** [review]
Litigious gentleman takes money

File: elements/rh-audio-player/rh-audio-player.ts
Comment (nikkimk) [inline elements/rh-audio-player/rh-audio-player.ts:32]: We importing `rh-tooltip` in line 32 if we already did in line 1?

Comment (bennypowers) [reply elements/rh-audio-player/rh-audio-player.ts:32]: b/c of type imports vs runtime

**Comment by zeroedin** [comment]
> @zeroedin @bennypowers the playground demo is broken for button

This looks to be due to `@lit/context` not being available in the importmap.  Added it however looks like there is another issue that follows.

File: package.json
Comment (bennypowers) [reply package.json]: on second thought, i have to revise my previous comment. we still rely on `<pf-icon>` e.g. in footer, so for now we'll keep the dependency, but the only imports should be pf-icon, really

**Comment by bennypowers** [comment]
If I'm not mistaken (and I hope I am) this is blocked on https://github.com/lit/lit/issues/4609

If I'm right, we can unblock ourselves with a workaround that adopts Lit's particular interpretation of the context protocol. That's arguably a breaking change though

**Comment by zeroedin** [comment]
I reverted the prior change to `rh-tabs` which pulls in `@lit/context`.  I think that would at least unblock us to upgrade deps.  We can come back around to upgrading tabs when we have the fix for @lit/context

**Comment by zeroedin** [comment]
> I reverted the prior change to `rh-tabs` which pulls in `@lit/context`. I think that would at least unblock us to upgrade deps. We can come back around to upgrading tabs when we have the fix for @lit/context

FYI: Will be reverting this again with a force push.  We can resolve with the latest released version of [`@lit/context`](https://github.com/lit/lit/releases/tag/%40lit%2Fcontext%401.1.1)

File: elements/rh-menu/rh-menu.ts
Comment (bennypowers) [inline elements/rh-menu/rh-menu.ts:38]: No side effects in getters, please

**Pull request merged**
